tests/test_json_schema.py

```diff
@@ -0,0 +1,258 @@
+"""
+Test cases (valid and invalid ones) to test the JSON schema.
+
+Run tests with `pytest test_json_schema.py`
+
+Test cases are generated on the fly, but if you want to have them written
+as .json files to inspect, run `python test_json_schema.py`
+
+"""
+import copy
+import json
+import pathlib
+
+from jsonschema.validators import Draft7Validator
+
+import pytest
+
+
+HERE = pathlib.Path(__file__).parent
+SCHEMA_SRC = HERE / ".." / "format-specs" / "schema.json"
+SCHEMA = json.loads(SCHEMA_SRC.read_text())
+
+
+# # Define test cases
+
+valid_cases = {}
+invalid_cases = {}
+
+
+metadata_template = {
+    "version": "0.5.0-dev",
+    "primary_column": "geometry",
+    "columns": {
+        "geometry": {
+            "encoding": "WKB",
+            "geometry_types": [],
+        },
+    },
+}
+
+
+# Minimum required metadata
+
+metadata = copy.deepcopy(metadata_template)
+valid_cases["minimal"] = metadata
+
+metadata = copy.deepcopy(metadata_template)
+metadata.pop("version")
+invalid_cases["missing_version"] = metadata
+
+metadata = copy.deepcopy(metadata_template)
+metadata.pop("primary_column")
+invalid_cases["missing_primary_column"] = metadata
+
+metadata = copy.deepcopy(metadata_template)
+metadata.pop("columns")
+invalid_cases["missing_columns"] = metadata
+
+metadata = copy.deepcopy(metadata_template)
+metadata["columns"] = {}
+invalid_cases["missing_columns_entry"] = metadata
```

Comment:
This one is still failing (it is xfailed for now to have the tests passing):

```
E           AssertionError: This is an invalid GeoParquet file, but no validation error occurred for 'missing_columns_entry':
E           {
E             "geo": {
E               "columns": {},
E               "primary_column": "geometry",
E               "version": "0.5.0-dev"
E             }
E           }
```

But I suppose that should be easy to solve to require a minimum length of one for `columns`?

Reply:
Ah, I see. We can fix that, sure. Just add `"minProperties": 1,` in the columns schema.
By the way, the written spec doesn't forbid an empty column object.

Reply:
Example:

```json
    "columns": {
      "type": "object",
      "minProperties": 1,
      "patternProperties": {
        ".+": {

```